### PR TITLE
Feat/no frontrunning

### PR DESCRIPTION
**Pull request by oscgu**
~~Not to be merged yet, tests not working as of right now~~

Tests all pass

### CODE REVIEW

File: src/PRS.sol [hunk 1]
```diff
@@ -164,27 +170,60 @@ contract PRS is Ownable, Pausable, TaxableGame {
...
         if (game.p2 == address(0)) revert Errors.NoSecondPlayer();
-        if (game.p2 != msg.sender) revert Errors.NotSecondPlayer(game.p2, msg.sender);
-        if (!_didTimerRunOut(game.timerStart)) revert Errors.TimerStillRunning();
 
-        _payout(msg.sender, game.entryFee*2);
+        bool isTimerRunning = !_didTimerRunOut(game.timerStart);
+        bool isP1ChoiceNone = game.p1ClearChoice == Choices.NONE;
+        bool isP2ChoiceNone = game.p2ClearChoice == Choices.NONE;
+
+        // @notice Game is not resolvable is timer is still running and both players have not revealed their move
```
Comment (faces-of-eth) [inline src/PRS.sol:207]: "Game is not resolvable _if_ timer…"

File: src/PRS.sol [hunk 2]
```diff
@@ -164,27 +170,60 @@ contract PRS is Ownable, Pausable, TaxableGame {
...
+        if (!isTimerRunning && isP2ChoiceNone && !isP1ChoiceNone) {
+            _payout(p1, entryFee);
+            return;
+        }
+
+        // @notice Timer ran out and only p1 did not reveal
+        if (!isTimerRunning && isP1ChoiceNone && isP2ChoiceNone) {
+            _payout(game.p2, entryFee);
+            return;
+        }
```
Comment (faces-of-eth) [inline src/PRS.sol:232]: For consistency's sake I'd structure the if statements the same way.

Have the same variables in the same order, just swap the !

Comment (faces-of-eth) [reply src/PRS.sol:232]: Actually could you do both like this…remember, I tweaked `_payout` to take full amount, so it needs the full game amount.

```
if (!isTimerRunning && !isP1ChoiceNone && isP2ChoiceNone) {
  _payout(p1, entryFee * 2);
  return;
}

if (!isTimerRunning && isP1ChoiceNone && !isP2ChoiceNone) {
  _payout(p2, entryFee * 2);
  return;
}
```

**Review by faces-of-eth — DISMISSED** [review]

File: src/PRS.sol (hunk 2 above)
Comment (oscgu) [reply src/PRS.sol]: Check line 206, should change the name to reflect better what is being paid out

**Pull request merged**